### PR TITLE
objects: don't recompute object size

### PR DESCRIPTION
**Pull request by efiop**
Was discovered as a part of https://github.com/iterative/dvc/issues/7824

### CODE REVIEW

File: scmrepo/git/backend/pygit2.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 import os
 import stat
 from contextlib import contextmanager
+from functools import cached_property
```
Comment (skshetry) [inline scmrepo/git/backend/pygit2.py:6]: I'll suggest sticking to `funcy.cached_property` instead due to https://github.com/python/cpython/issues/87634.

Unless we are trying to replace the following piece of code:
```python
@funcy.wrap_prop(threading.RLock())
@funcy.cached_property

# equivalent to @functools.cached_property
```

Comment (efiop) [reply scmrepo/git/backend/pygit2.py:6]: Makes sense, was just trying it out with pylint and 3.8

Comment (efiop) [reply scmrepo/git/backend/pygit2.py]: Didn't know about that bug though 🙁 Thank you!

File: scmrepo/git/backend/pygit2.py [hunk 2]
```diff
@@ -62,8 +62,8 @@ def mode(self):
             return stat.S_IFDIR
         return self.obj.filemode
 
-    @property
-    def size(self) -> int:
+    @cached_property
+    def size(self) -> int:  # pylint: disable=invalid-overridden-method
         try:
             return len(self.obj.read_raw())
```
Comment (efiop) [inline scmrepo/git/backend/pygit2.py:68]: Welp, actually there is size, but just not for trees. Will send a followup.

**Pull request merged**